packages/engine/src/simulation/package/context/packages/api_requests/mod.rs

```diff
@@ -88,33 +89,21 @@ impl Package for ApiRequests {
         state: Arc<State>,
         snapshot: Arc<StateSnapshot>,
     ) -> Result<Vec<ContextColumn>> {
-        let mut api_response_maps = if let Some(ref handlers) = self.custom_message_handlers {
-            let mut futs = FuturesOrdered::new();
-            {
-                let message_pool = snapshot.message_pool();
-                let message_pool_read = message_pool
-                    .read()
-                    .map_err(|e| Error::from(e.to_string()))?;
-                let reader = message_pool_read.get_reader();
-
-                handlers.iter().try_for_each::<_, Result<()>>(|handler| {
-                    let messages = snapshot.message_map().get_msg_refs(handler);
-                    if !messages.is_empty() {
-                        let messages = handlers::gather_requests(&reader, messages)?;
-                        futs.push(handlers::run_custom_message_handler(handler, messages))
-                    }
-                    Ok(())
-                })?;
-            }
+        // We want to pass the span for the package to the writer, so that the write() call isn't
+        // nested under the run span
+        let pkg_span = Span::current();
+        let run_span = tracing::trace_span!("run"); // store an un-entered span for the async
```

Comment:
```suggestion
        // store an un-entered span for the async
        let run_span = tracing::trace_span!("run");
```

Reply:
Have chatted about this offline, differing opinion about inline comments